test/hotspot/jtreg/runtime/DefineClass/NullClassBytesTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8262913
+ * @summary Verifies DefineClass with null or truncate bytes gets appropriate exception
+ * @library /test/lib
+ * @modules java.base/jdk.internal.misc
+ * @compile A.java
+ * @run main/native NullClassBytesTest
+ */
+
+import java.io.*;
+
+public class NullClassBytesTest {
+
+    static native Class<?> nativeDefineClass(String name, ClassLoader ldr, byte[] class_bytes, int length);
+
+    static {
+        System.loadLibrary("NullClassBytesTest");
+    }
+
+    static class SimpleLoader extends ClassLoader {
+
+        public Class<?> loadClass(String name) throws ClassNotFoundException {
+            synchronized(getClassLoadingLock(name)) {
+                Class<?> c = findLoadedClass(name);
+                if (c != null) return c;
+
+                // load the class data from the connection
+                if (name.equals("A")) {
+                    byte[] b = getClassData("A");
+                    return defineClass(name, b, 0, b.length);
+                } else if (name.equals("B")) {
+                    byte[] b = new byte[0];
+                    return defineClass(name, b, 0, b.length);
+                } else if (name.equals("C")) {
+                    byte[] b = null;
+                    return defineClass(name, b, 0, 0);
+                } else if (name.equals("D")) {
+                    byte[] b = new byte[0];
+                    return nativeDefineClass(name, this, b, b.length);
+                } else if (name.equals("E")) {
+                    byte[] b = null;
+                    return nativeDefineClass(name, this, b, 0);
+                } else {
+                    return super.loadClass(name);
+                }
+            }
+        }
+
+        byte[] getClassData(String name) {
```

Comment:
This function can be simplified to something like

```
return SimpleLoader.class.getClassLoader().getResourceAsStream(name + ".class").readAllBytes();
```

That way it will be more resilient if jtreg wants to put the classfile elsewhere.

Reply:
I like this suggestion!  We should file an RFE to fix all the tests that have this getClassData function.

Reply:
I think this was supposed to be SimpleLoader.getResourceAsStream(name + ".class").readAllBytes();